0004/README.md

```diff
@@ -0,0 +1,41 @@
+---
+id: 0004
+title: Tinkerbell Security
+status: ideation
+authors: chitrabasu khare <chitrabasukhare89@gmail.com>
+---
+
+## Summary:
+
+Currently, all the services in provisioner are accessible from outside (i.e via worker). Even in some implementation, ex tink #195, worker have full access (Read+write) tink-server via tink client. This leads to the possibility that worker can alter the behaviour of tink-server.
+
+## Goals:
+
+- Move all Tinkerbell services in private Network and expose them via proxy on a given port.
```

Comment:
I generally find that I am looking for more context in this proposal.  I'll use the goal as an example since the context I am looking for around this goal would have a strong influence on the content and perhaps the summary too.

> Move all Tinkerbell services in private Network

In what scenario can we assume that tinkerbell has not already been installed into private-only network?  Does this proposal recommend changes to the default installer options or the services in general?

I think this implies that we will be changing the installation defaults. We offer 3 different installers. Will we be using a common address in all of these environments (I assume not)? If not, what principle will be used to define the private network (options, preferred interface (eth1?), preferred rfc-1918 range (10.x) )?

I would like to see this proposal add more detail about how this goal could be achieved.
Defining a common set of microservice parameters, such as `--bind-address 0.0.0.0 --bind-port 1234` or `--bind-address ::1`, for example, allows the user to decide what makes a private or protected network.

> expose them via proxy on a given port

Likewise, this could be offered as a `--proxy-address` and `--proxy-port`.  Would we get the existing behavior without these parameters? Would the default `proxy-address` be the `bind-address`?  What services (specifically) would be proxied?


Documenting these parameters and instrumenting our installers to make private addresses the default (perhaps behind a toggle) would be secondary goals.